### PR TITLE
Mini-Challenge 4

### PR DESCRIPTION
**Pull request by sofisa16**
- The search term is stored and retrieved from the Global Context correctly.
- The appearance theme is stored on the Global Context and applied correctly to the App UI.
- Using useReducer for adding fav videos.

### CODE REVIEW

File: src/components/FavoritesView/addToFavorites.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export const initialState: FavoriteVideos = {};
+
+export interface Action {
+  type: string;
+  payload: YouTubeResponseItems;
+}
+
+export function reducer(state: FavoriteVideos, action: Action): FavoriteVideos {
+  const id = typeof(action.payload.id) === 'string' ? action.payload.id : action.payload.id && action.payload.id.videoId;
```
Comment (MartinMolinero) [inline src/components/FavoritesView/addToFavorites.tsx:15]:
```suggestion
  const id = typeof(action.payload.id) === 'string' ? action.payload.id : action.payload?.id?.videoId;
```

File: src/__tests__/components/Header/components/SearchBox/SearchBox.test.tsx [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+
+  const tree = renderer
+    .create(
+      <Router history={history}>
+        <SearchBox />
+      </Router>
+    )
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (MartinMolinero) [inline src/__tests__/components/Header/components/SearchBox/SearchBox.test.tsx:37]: You can also add the test to check that after the user writes inside the box, the text is present in the element.

File: src/components/VideoDetailsView/VideoDetailsView.tsx [hunk 3]
```diff
@@ -65,10 +69,12 @@ function VideoDetailsView(): JSX.Element {
   //const related: YouTubeResponse = bug;
   const [embedHtml, setEmbedHtml] = useState<string>('');
   const [relatedVideos, setRelatedVideos] = useState<JSX.Element[]>([]);
+  const {favoriteVideos, dispatchFav} = useContext(GlobalContext);
+  const [buttonLabel, setButtonLabel] = useState<string>('Agregar a favoritos');
 
   useEffect(
     () => {
-      async function getVideoFromYoutube(videoId: string): Promise<void> {
+      async function getVideoFromYoutube(/*videoId: string*/): Promise<void> {
```
Comment (MartinMolinero) [inline src/components/VideoDetailsView/VideoDetailsView.tsx:77]: why were this commented out?

File: src/components/Mosaic/Mosaic.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+function Mosaic(props: MosaicProps): JSX.Element {
+  const {items} = props;
+  const [cards, setCards] = useState<JSX.Element[]>([]);
+
+  useEffect(
+    () => {
+      const cards: JSX.Element[] = [];
+      for(const item of items) {
```
Comment (MartinMolinero) [inline src/components/Mosaic/Mosaic.tsx:24]: verify the case in which they can be undefined, this may cause some crashes in your app

**Review by MartinMolinero — COMMENTED** [review]
I have a couple of suggestions for your project. I can see you added logic to handle the open and close action of the drawer as well as the dark mode, but in your demo url i can't see it. Please update it so I can have a look at the functionality as well. The code looks good.

File: src/components/VideoDetailsView/VideoDetailsView.tsx (hunk 3 above)
Comment (sofisa16) [reply src/components/VideoDetailsView/VideoDetailsView.tsx]: I forgot to uncomment them. Thank you!

**Pull request merged**